src/list_store.rs

```diff
@@ -21,6 +21,30 @@ impl ListStore {
         }
     }
 
+    pub fn insert_with_values(&self, position: Option<u32>, columns: &[u32], values: &[&ToValue])
+            -> TreeIter {
+        unsafe {
+            assert!(position.unwrap_or(0) <= i32::max_value() as u32);
+            let n_columns = ffi::gtk_tree_model_get_n_columns(self.to_glib_none().0) as u32;
+            assert!(columns.len() == values.len());
+            assert!(columns.len() <= n_columns as usize);
+            for &column in columns {
+                assert!(column < n_columns);
+                let type_ = from_glib(
+                    ffi::gtk_tree_model_get_column_type(self.to_glib_none().0, column as c_int));
+                assert!(Value::type_transformable(values[column as usize].to_value_type(), type_));
```

Comment:
`assert_eq!` is much more expensive than `assert!`, I wouldn't use it outside of tests.